src/build/arg/mod.rs

```diff
@@ -2720,27 +2720,56 @@ impl<'help> Arg<'help> {
+    /// [`ArgMatches::occurrences_of`] will return `0`.
+    /// [`ArgMatches::is_present`] will return `true` if the variable is present
+    /// in the environment . If you wish to determine whether the argument was
+    /// used at runtime or not, consider [`ArgMatches::occurrences_of`] which
+    /// will return `0` if the argument was *not* used at runtime.
+    ///
+    /// If [`Arg::multiple(true)`] is set then [`Arg::use_delimiter(true)`]
+    /// should also be set. Otherwise, only a single argument will be returned
+    /// from the environment variable. The default delimiter is `,` and follows
+    /// all the other delimiter rules.
+    ///
+    /// If user sets the argument in the environment:
+    /// - When [`Arg::takes_value(true)`] is not set, the flag is considered raised.
+    /// - When [`Arg::takes_value(true)`] is set, [`ArgMatches::value_of`] will
+    ///   return value of the environment variable.
+    ///
+    /// If user doesn't set the argument in the environment:
+    /// - When [`Arg::takes_value(true)`] is not set, the flag is considered off.
+    /// - When [`Arg::takes_value(true)`] is set, [`ArgMatches::value_of`] will
+    ///   return the default specified.
     ///
-    /// **NOTE:** If the user *does not* use this argument at runtime, [`ArgMatches::occurrences_of`]
-    /// will return `0` even though the [`ArgMatches::value_of`] will return the default specified.
+    /// # Examples
     ///
-    /// **NOTE:** If the user *does not* use this argument at runtime [`ArgMatches::is_present`] will
-    /// return `true` if the variable is present in the environment . If you wish to determine whether
-    /// the argument was used at runtime or not, consider [`ArgMatches::occurrences_of`] which will
-    /// return `0` if the argument was *not* used at runtime.
+    /// In this example, we show the variable coming from the environment:
     ///
-    /// **NOTE:** This implicitly sets [`Arg::takes_value(true)`].
+    /// ```rust
+    /// # use std::env;
+    /// # use clap::{App, Arg};
     ///
-    /// **NOTE:** If [`Arg::multiple(true)`] is set then [`Arg::use_delimiter(true)`] should also be
-    /// set. Otherwise, only a single argument will be returned from the environment variable. The
-    /// default delimiter is `,` and follows all the other delimiter rules.
+    /// env::set_var("MY_FLAG", "env");
     ///
-    /// # Examples
+    /// let m = App::new("prog")
+    ///     .arg(Arg::new("flag")
+    ///         .long("flag")
+    ///         .env("MY_FLAG")
+    ///         .takes_value(true))
+    ///     .get_matches_from(vec![
+    ///         "prog"
+    ///     ]);
     ///
-    /// In this example, we show the variable coming from the environment:
+    /// assert_eq!(m.value_of("flag"), Some("env"));
+    /// ```
+    ///
+    /// In this example, we cannot show the variable coming from the environment
+    /// without setting [`Arg::takes_value(true)`]:
```

Comment:
In this example, we show the flag being raised but with no value because of not setting Arg::takes_values(true)